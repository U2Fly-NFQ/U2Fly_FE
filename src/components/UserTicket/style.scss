@use '../../assets/scss/abstracts' as a;

.userTicket {
  width: 1000px !important;
  padding-bottom: 0;

  .ant-modal-content {
    border-radius: 25px;

    .ant-modal-body {
      padding: 0;
    }
  }

  .ticket {
    img {
      width: 100%;
    }

    &-header {
      text-align: center;
      align-items: center;
      padding: 15px;
      background-color: a.$bg-color;
      border-top-left-radius: 25px;
      border-top-right-radius: 25px;
    }

    &-content {
      align-items: center;
      .passenger-info-title,
      .flight-details-title {
        font-size: 16px;
      }

      .passenger-info-text,
      .flight-details-text {
        font-weight: bold;
        font-size: 18px;
      }

      .flight-info {
        align-items: center;
        &-from {
          text-align: center;
        }

        &-icon {
          text-align: center;
        }

        &-to {
          text-align: center;
        }
      }

      .info_empty:after {
        content: '';
        border-bottom: 1px dashed a.$main-color;
        position: absolute;
        left: 0;
        right: 0;
        bottom: 10px;
      }

      &-left {
        padding: 15px 0;
        & > div {
          align-items: center;
        }

        &-barcode {
          img {
            transform: rotate(-90deg);
          }
        }

        .passenger-info {
          text-align: center;
        }

        .flight-info {
          padding: 40px 0;

          &-from,
          &-icon,
          &-to {
            font-weight: bold;
            font-size: 40px;
          }
        }

        .flight-details {
          text-align: center;
        }
      }

      &-right {
        padding: 15px 0;
        padding-right: 15px;
        .barcode {
          img {
            max-height: 75px;
          }
        }

        .passenger-info > div > div {
          margin-bottom: 5px;
        }

        .flight-info {
          font-weight: bold;
          font-size: 20px;
        }
      }
    }

    &-footer {
      background-color: a.$bg-color;
      border-bottom-left-radius: 25px;
      border-bottom-right-radius: 25px;
      & > div {
        padding: 10px;
      }
      &-left {
        text-align: center;
      }
    }
    &:after {
      content: '';
      top: 0;
      position: absolute;
      border: 1px a.$main-color dashed;
      bottom: 0;
      left: 68.5%;
    }
  }
}

@media only screen and (max-width: 1000px) {
  .userTicket {
    width: 900px !important;

    .ticket {
      zoom: 0.9;
    }
  }
}

@media only screen and (max-width: 900px) {
  .userTicket {
    width: 800px !important;

    .ticket {
      zoom: 0.8;
    }
  }
}

@media only screen and (max-width: 800px) {
  .userTicket {
    width: 700px !important;

    .ticket {
      zoom: 0.7;
    }
  }
}

@media only screen and (max-width: 700px) {
  .userTicket {
    width: 600px !important;

    .ticket {
      zoom: 0.6;
    }
  }
}

@media only screen and (max-width: 600px) {
  .userTicket {
    width: 500px !important;

    .ticket {
      zoom: 0.5;
    }
  }
}

@media only screen and (max-width: 500px) {
  .userTicket {
    width: 400px !important;

    .ticket {
      zoom: 0.4;
    }
  }
}

@media only screen and (max-width: 400px) {
  .userTicket {
    width: 300px !important;

    .ticket {
      zoom: 0.3;
    }
  }
}

@media only screen and (max-width: 319px) {
  .userTicket {
    display: none;
  }
}
