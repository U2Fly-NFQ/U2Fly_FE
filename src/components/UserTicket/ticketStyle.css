@import url('https://fonts.googleapis.com/css2?family=Oswald:wght@200;400;500;600;700&display=swap');

.ticket {
  width: 100%;
  border-radius: 30px;
  box-shadow: 1px 1px 11px 0 rgba(50, 50, 50, 0.75);
  display: flex;
  align-items: center;
  justify-content: space-around;
  position: relative;
  overflow: hidden;
  background-color: #f7f7f7;
}

.ticket::after {
  content: '';
  position: absolute;
  top: 0;
  width: 100%;
  height: 70px;
  background-color: #487261;
}

#barBig {
  position: relative;
  z-index: 2;
  width: 80px;
  height: 300px;
  /* padding: 30px; */
  /* margin: 30px; */
}

#barBig::after {
  content: '';
  position: absolute;
  bottom: 110px;
  right: -110px;
  background-image: url(https://www.incimages.com/uploaded_files/image/1024x576/*Barcode_32896.jpg);
  transform: rotate(90deg);
  background-size: cover;
  height: 80px;
  width: 300px;
}

#barBig img {
  transform: rotate(90deg);
  height: 100%;
  width: 100%;
}

#mainInfo {
  width: 50%;
  height: 100%;
  padding: 5px;
  padding-top: 35px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.passenger-info-container,
.flight-info-container {
  position: relative;
  z-index: 3;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-weight: 500;
  width: 100%;
}

.passenger-info,
.flight-info,
.flight-details div {
  display: flex;
  flex-direction: column;
}

.passenger-info span:nth-of-type(1) {
  color: black;
  font-weight: bold;
  font-size: 1.3rem;
}

.passenger-info span:nth-of-type(2) {
  margin-top: 2px;
  font-size: 1.6rem;
}

.destination {
  display: flex;
  align-items: center;
  font-size: 3rem;
}

.destination img {
  width: 50px;
  transform: rotate(45deg);
  margin: 0 30px;
}

.flight-info {
  align-items: center;
}

.flight-info span {
  font-size: 1.1rem;
}

.flight-info span:nth-of-type(1) {
  color: red;
}

#sideInfo {
  width: 30%;
  height: calc(100% - 80px);
  display: flex;
  flex-direction: column;
  align-items: center;
  align-self: flex-end;
}

#circles {
  /* border-left:10px dotted black; */
  height: 100%;
  z-index: 3;
  width: 1%;
}

.circle {
  width: 13px;
  height: 13px;

  background: #f8fcf0;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.5);
  border-radius: 50%;
  margin: 7px 0;
}

#barSmall {
  width: 220px;
  height: 50px;
  object-fit: cover;
}

.details-container {
  display: flex;
  justify-content: space-between;
  width: 210px;
}

.flight-details div span:nth-of-type(1) {
  color: #a5a5a5;
  font-size: 1.1rem;
}

.flight-details div span:nth-of-type(2) {
  font-size: 1.4rem;
  font-weight: bold;
}

.destination-details {
  display: flex;
  justify-content: center;
  align-items: center;
  transform: rotate(-90deg);
  font-size: 1.5rem;
  font-weight: bold;
  /* height:200px; */
  width: 30px;
}

.destination-details img {
  transform: rotate(45deg);
  height: 30px;
  margin-right: 10px;
  margin-left: 10px;
}
