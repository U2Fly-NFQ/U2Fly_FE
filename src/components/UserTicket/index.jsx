import React from 'react'
import { Col, Modal, Row } from 'antd'
import './style.scss'
import moment from 'moment'

function UserTicket({ visible, setViewTicket, ticketData }) {
  return (
    <Modal
      className="userTicket"
      visible={visible}
      footer={false}
      onCancel={() => setViewTicket(false)}
    >
      <div className="ticket">
        <Row justify="space-between" className="ticket-header">
          <Col span={8} className="ticket-header-logo">
            <img src={ticketData.flights.airline.image} alt="logo" />
          </Col>
          <Col span={8} className="ticket-header-class"></Col>
          <Col span={7} className="ticket-header-right">
            <img src={ticketData.flights.airline.image} alt="logo" />
          </Col>
        </Row>

        <Row justify="space-between" className="ticket-content">
          <Col span={16} className="ticket-content-left">
            <Row gutter={8}>
              <Col className="ticket-content-left-barcode" span={4}>
                <img
                  src="https://cdn.pixabay.com/photo/2014/04/02/16/19/barcode-306926__480.png"
                  alt="barcode"
                />
              </Col>
              <Col className="ticket-content-left-info" span={20}>
                <Row justify="space-between" className="passenger-info">
                  <Col span={8}>
                    <div className="passenger-info-title">
                      Full Name / Họ và Tên
                    </div>
                    <div className="passenger-info-text">
                      {ticketData.passenger.name}
                    </div>
                  </Col>
                  <Col span={6}>
                    <div className="passenger-info-title">
                      Flight/Chuyến Bay
                    </div>
                    <div className="passenger-info-text">
                      {ticketData.flights.code}
                    </div>
                  </Col>
                  <Col span={4}>
                    <div className="passenger-info-title">Date/Ngày</div>
                    <div className="passenger-info-text">
                      {moment(ticketData.flights.startDate).format('DD/M/YYYY')}
                    </div>
                  </Col>
                  <Col span={3} className="info_empty">
                    <div className="passenger-info-title">Seat/Ghế</div>
                    <div className="passenger-info-text">&nbsp;</div>
                  </Col>
                </Row>
                <Row className="flight-info">
                  <Col span={10} className="flight-info-from">
                    {ticketData.flights.departure}
                  </Col>
                  <Col span={4} className="flight-info-icon">
                    <i className="fa-solid fa-plane"></i>
                  </Col>
                  <Col span={10} className="flight-info-to">
                    {ticketData.flights.arrival}
                  </Col>
                </Row>
                <Row justify="space-between" className="flight-details">
                  <Col span={5} className="info_empty">
                    <div className="flight-details-title">Gate/Cổng</div>
                    <div className="flight-details-text">&nbsp;</div>
                  </Col>
                  <Col span={5}>
                    <div className="flight-details-title">Boarding Time</div>
                    <div className="flight-details-text">
                      {ticketData.flights.boardingTime}
                    </div>
                  </Col>
                  <Col span={5}>
                    <div className="flight-details-title">ETD/Khởi Hành</div>
                    <div className="flight-details-text">
                      {ticketData.flights.ETD}
                    </div>
                  </Col>
                  <Col span={5}>
                    <div className="flight-details-title">ETA/Đến Nơi</div>
                    <div className="flight-details-text">
                      {ticketData.flights.ETA}
                    </div>
                  </Col>
                </Row>
              </Col>
            </Row>
          </Col>
          <Col span={7} className="ticket-content-right">
            <Row>
              <Col span={24} className="barcode">
                <img
                  src="https://cdn.pixabay.com/photo/2014/04/02/16/19/barcode-306926__480.png"
                  alt="barcode"
                />
              </Col>
              <Col span={24}>
                <Row className="flight-info">
                  <Col span={10} className="flight-info-from">
                    {ticketData.flights.departure}
                  </Col>
                  <Col span={4} className="flight-info-icon">
                    <i className="fa-solid fa-plane"></i>
                  </Col>
                  <Col span={10} className="flight-info-to">
                    {ticketData.flights.arrival}
                  </Col>
                </Row>
              </Col>
              <Col span={24} className="passenger-info">
                <Row>
                  <Col span={24}>
                    <div className="passenger-info-title">
                      Full Name / Họ và Tên
                    </div>
                    <div className="passenger-info-text">
                      {ticketData.passenger.name}
                    </div>
                  </Col>
                  <Col span={24}>
                    <div className="passenger-info-title">
                      Flight/Chuyến Bay
                    </div>
                    <div className="passenger-info-text">
                      {ticketData.flights.code}
                    </div>
                  </Col>
                  <Col span={24}>
                    <div className="passenger-info-title">Date/Ngày</div>
                    <div className="passenger-info-text">
                      {moment(ticketData.flights.startDate).format('DD/M/YYYY')}
                    </div>
                  </Col>
                  <Col span={24} className="info_empty">
                    <div className="passenger-info-title">Seat/Ghế</div>
                    <div className="passenger-info-text">&nbsp;</div>
                  </Col>
                </Row>
              </Col>
            </Row>
          </Col>
        </Row>

        <Row justify="space-between" className="ticket-footer">
          <Col span={16} className="ticket-footer-left">
            Gate closes 20 minutes before departure - Cổng đóng 20 phút trước
            khi khởi hành
          </Col>
          <Col span={7} className="ticket-footer-right">
            Hotline: 19XX XXX XXX
          </Col>
        </Row>
      </div>
    </Modal>
  )
}

export default UserTicket
