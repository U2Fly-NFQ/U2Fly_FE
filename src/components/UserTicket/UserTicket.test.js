import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import UserTicket from './'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../translations'

const mockStore = configureStore([])

describe('My Connected React-Redux Component', () => {
  let store
  let component

  component = renderer.create(
    <I18nextProvider i18n={i18n}>
      <UserTicket
        visible={true}
        setViewTicket={() => {}}
        ticketData={{
          passenger: { name: 'John' },
          flights: { airline: { image: '' } },
        }}
      />
    </I18nextProvider>
  )

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
