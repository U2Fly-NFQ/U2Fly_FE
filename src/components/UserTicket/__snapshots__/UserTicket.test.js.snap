// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="ant-modal-root"
>
  <div
    className="ant-modal-mask"
    style={
      Object {
        "zIndex": undefined,
      }
    }
  />
  <div
    className="ant-modal-wrap"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "display": null,
        "zIndex": undefined,
      }
    }
    tabIndex={-1}
  >
    <div
      aria-labelledby={null}
      aria-modal="true"
      className="ant-modal userTicket"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "width": 520,
        }
      }
    >
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-modal-content"
      >
        <button
          aria-label="Close"
          className="ant-modal-close"
          onClick={[Function]}
          type="button"
        >
          <span
            className="ant-modal-close-x"
          >
            <span
              aria-label="close"
              className="anticon anticon-close ant-modal-close-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          className="ant-modal-body"
        >
          <div
            className="ticket"
          >
            <div
              className="ant-row ant-row-space-between ticket-header"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-8 ticket-header-logo"
                style={Object {}}
              >
                <img
                  alt="logo"
                  src=""
                />
              </div>
              <div
                className="ant-col ant-col-8 ticket-header-class"
                style={Object {}}
              />
              <div
                className="ant-col ant-col-7 ticket-header-right"
                style={Object {}}
              >
                <img
                  alt="logo"
                  src=""
                />
              </div>
            </div>
            <div
              className="ant-row ant-row-space-between ticket-content"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-16 ticket-content-left"
                style={Object {}}
              >
                <div
                  className="ant-row"
                  style={
                    Object {
                      "marginLeft": -4,
                      "marginRight": -4,
                    }
                  }
                >
                  <div
                    className="ant-col ant-col-4 ticket-content-left-barcode"
                    style={
                      Object {
                        "paddingLeft": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <img
                      alt="barcode"
                      src="https://cdn.pixabay.com/photo/2014/04/02/16/19/barcode-306926__480.png"
                    />
                  </div>
                  <div
                    className="ant-col ant-col-20 ticket-content-left-info"
                    style={
                      Object {
                        "paddingLeft": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <div
                      className="ant-row ant-row-space-between passenger-info"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-8"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Full Name / Họ và Tên
                        </div>
                        <div
                          className="passenger-info-text"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-7"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Flight/Chuyến Bay
                        </div>
                        <div
                          className="passenger-info-text"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-5"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Date/Ngày
                        </div>
                        <div
                          className="passenger-info-text"
                        >
                          07-12-2022
                        </div>
                      </div>
                      <div
                        className="ant-col ant-col-4 info_empty"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Seat/Ghế
                        </div>
                        <div
                          className="passenger-info-text"
                        >
                           
                        </div>
                      </div>
                    </div>
                    <div
                      className="ant-row flight-info"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-10 flight-info-from"
                        style={Object {}}
                      />
                      <div
                        className="ant-col ant-col-4 flight-info-icon"
                        style={Object {}}
                      >
                        <i
                          className="fa-solid fa-plane"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-10 flight-info-to"
                        style={Object {}}
                      />
                    </div>
                    <div
                      className="ant-row ant-row-space-between flight-details"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-4 info_empty"
                        style={Object {}}
                      >
                        <div
                          className="flight-details-title"
                        >
                          Gate/Cổng
                        </div>
                        <div
                          className="flight-details-text"
                        >
                           
                        </div>
                      </div>
                      <div
                        className="ant-col ant-col-6"
                        style={Object {}}
                      >
                        <div
                          className="flight-details-title"
                        >
                          Boarding Time
                        </div>
                        <div
                          className="flight-details-text"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-6"
                        style={Object {}}
                      >
                        <div
                          className="flight-details-title"
                        >
                          ETD/Khởi Hành
                        </div>
                        <div
                          className="flight-details-text"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-6"
                        style={Object {}}
                      >
                        <div
                          className="flight-details-title"
                        >
                          ETA/Đến Nơi
                        </div>
                        <div
                          className="flight-details-text"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-col ant-col-7 ticket-content-right"
                style={Object {}}
              >
                <div
                  className="ant-row"
                  style={Object {}}
                >
                  <div
                    className="ant-col ant-col-24 barcode"
                    style={Object {}}
                  >
                    <img
                      alt="barcode"
                      src="https://cdn.pixabay.com/photo/2014/04/02/16/19/barcode-306926__480.png"
                    />
                  </div>
                  <div
                    className="ant-col ant-col-24"
                    style={Object {}}
                  >
                    <div
                      className="ant-row flight-info"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-10 flight-info-from"
                        style={Object {}}
                      />
                      <div
                        className="ant-col ant-col-4 flight-info-icon"
                        style={Object {}}
                      >
                        <i
                          className="fa-solid fa-plane"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-10 flight-info-to"
                        style={Object {}}
                      />
                    </div>
                  </div>
                  <div
                    className="ant-col ant-col-24 passenger-info"
                    style={Object {}}
                  >
                    <div
                      className="ant-row"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-24"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Full Name / Họ và Tên
                        </div>
                        <div
                          className="passenger-info-text"
                        >
                          John
                        </div>
                      </div>
                      <div
                        className="ant-col ant-col-24"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Flight/Chuyến Bay
                        </div>
                        <div
                          className="passenger-info-text"
                        />
                      </div>
                      <div
                        className="ant-col ant-col-24"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Date/Ngày
                        </div>
                        <div
                          className="passenger-info-text"
                        >
                          07-12-2022
                        </div>
                      </div>
                      <div
                        className="ant-col ant-col-24 info_empty"
                        style={Object {}}
                      >
                        <div
                          className="passenger-info-title"
                        >
                          Seat/Ghế
                        </div>
                        <div
                          className="passenger-info-text"
                        >
                           
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-row ant-row-space-between ticket-footer"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-16 ticket-footer-left"
                style={Object {}}
              >
                Gate closes 20 minutes before departure - Cổng đóng 20 phút trước khi khởi hành
              </div>
              <div
                className="ant-col ant-col-7 ticket-footer-right"
                style={Object {}}
              >
                Hotline: 19XX XXX XXX
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;
